v4/newrelic/segments.go

```diff
@@ -295,10 +302,74 @@ func (s *ExternalSegment) End() {
 	if s.StartTime.isEnded() {
 		return
 	}
+	s.addAttributes(s.StartTime.Span.SetAttributes)
 	s.StartTime.Span.SetName(s.name())
+	s.setSpanStatus(s.StartTime.Span.SetStatus)
 	s.StartTime.end()
 }
 
+func (s *ExternalSegment) setSpanStatus(setter func(codes.Code, string)) {
+	var code int
+	if s.statusCode != nil {
+		code = *s.statusCode
+	} else if s.Response != nil {
+		code = s.Response.StatusCode
+	}
+	if code < 17 {
+		// Assume the code is already a grpc status code
+		c := codes.Code(code)
+		setter(c, c.String())
+		return
```

Comment:
this is purely stylistic, but I think I would prefer to not `return` here and instead use an `else` block below. I like early `return`s early on in a function when validating preconditions, but this late in the function I find it a little confusing.

Reply:
No problem! I'll make the change.